packages/@glimmerx/core/src/owner.ts

```diff
@@ -9,24 +9,37 @@ export interface FactoryIdentifier {
 
 const SERVICES = Symbol('Services');
 
+function isConstructor(func) {
+  return (
+    (func && typeof func === 'function' && func.prototype && func.prototype.constructor) === func
+  );
+}
+
 export default class Owner {
   private [SERVICES]: Dict<unknown>;
+  private delegate;
 
   constructor(services: Dict<unknown>) {
-    this[SERVICES] = services;
+    this[SERVICES] = {};
```

Comment:
We should likely just turn this into a `Map` so that the checks below can do `this[SERVICES].has(name)`.